src/Npgsql/NpgsqlSlimDataSourceBuilder.cs

```diff
@@ -477,6 +477,39 @@ public NpgsqlSlimDataSourceBuilder EnableIntegratedSecurity()
         return this;
     }
 
+    /// <summary>
+    /// Sets up network mappings. This allows mapping PhysicalAddress, IPAddress, NpgsqlInet and NpgsqlCidr types
+    /// to PostgreSQL <c>macaddr</c>, <c>macaddr8</c>, <c>inet</c> and <c>cidr</c> types.
+    /// </summary>
+    /// <returns>The same builder instance so that multiple calls can be chained.</returns>
+    public NpgsqlSlimDataSourceBuilder EnableNetworkTypes()
+    {
+        _resolverChainBuilder.AppendResolverFactory(new NetworkTypeInfoResolverFactory());
+        return this;
+    }
+
+    /// <summary>
+    /// Sets up network mappings. This allows mapping types like NpgsqlPoint and NpgsqlPath
+    /// to PostgreSQL <c>point</c>, <c>path</c> and so on types.
+    /// </summary>
+    /// <returns>The same builder instance so that multiple calls can be chained.</returns>
+    public NpgsqlSlimDataSourceBuilder EnableGeometricTypes()
+    {
+        _resolverChainBuilder.AppendResolverFactory(new GeometricTypeInfoResolverFactory());
+        return this;
+    }
+
+    /// <summary>
+    /// Sets up System.Text.Json mappings. This allows mapping JsonDocument and JsonElement types to PostgreSQL <c>json</c> and <c>jsonb</c>
+    /// types.
+    /// </summary>
+    /// <returns>The same builder instance so that multiple calls can be chained.</returns>
+    public NpgsqlSlimDataSourceBuilder EnableJson()
```

Comment:
Should we add Types there as well? It's not like this adds some general "Json support" to the driver beyond just supporting JsonDocument/JsonElement, right?

Reply:
It does create some inconsistency with `EnableDynamicJson`, not sure that matters enough. I could go either way.

Reply:
Yeah. If you squint hard, dynamic json doesn't enable support for any specific types so maybe it's ok ;)

My vote would be for EnableJsonTypes() here, but no strong feelings.

Reply:
Done